### DISCUSSION TITLE
Getting the user after sign up but before email confirmation

### DISCUSSION THREAD

**Discussion by namuorg** [supabase/supabase#29140] [Questions]
### Setup
Password-based authentication using `@supabase/ssr` with Next.js

### Scenario
1. User signs up
2. User takes too long to confirm their email and OTP expires
3. User returns to the application to try to sign in

### Questions

1. When the user returns to the login page, I want to automatically show a button to resend the confirmation email. However, it doesn't seem possible to get any information about the user since `auth.getUser` and `auth.getSession` returns data only after the user has signed in, which isn't allowed until the email is confirmed. As a result, it's not possible to automatically detect that they still need to confirm their email and show a "Resend confirmation" button. Is that correct?

2. When the user tries logging in without a confirmed email, the error that's returned has the message `"Email not confirmed"` with a error code of `undefined`. Is the only way to determine if the user hasn't confirmed their email to do a string comparison against the error message? That seems brittle (e.g. if Supabase changes the error message, the check would break). Is there a more reliable way to determine if the user needs to confirm their email? Since `user` is not available until the user has signed in, it's not possible to check `user.confirmed_at`.

I'm confused by the design decision to prevent the user from logging in unless their email is confirmed because it makes it impossible to keep track of the user after they've signed up but before they've confirmed their email. Firebase Auth doesn't have this issue because users can be signed in immediately after signing up, even when their email is not yet confirmed. Turning off "Confirm email" in Supabase's project settings disables email confirmations entirely so it doesn't address this problem.

I'd appreciate any clarifications if I'm missing something and there's actually a way to get the user state before they've confirmed their email. Thanks!

**Comment by DevEchoStorm** [comment]
facing exact same issue

**Comment by diogocapela** [comment]
I have the same issue. Any news?
